### PR TITLE
加入新的前端及優化後端

### PR DESCRIPTION
**Pull request by bell-rabbit**
Add a new UI.

Feature:
* 重新設計全個UI
* 保存最近10次的記錄
* 重新設計角色icon 顯示
* 同一個池下的角色出現率報告
* API加入TEMP的功能

pr_1
pr_2
pr_3
pr_4

### CODE REVIEW

File: flipper_gacha.py [hunk 1]
```diff
@@ -58,126 +64,167 @@ def limit_key_func():
     "halfanv": {"name": "半周年禮黑", "type": "three_pu"},
 }
 
+# Add character temp
+character_list_temp = {}
```
Comment (Zhen-Bo) [inline flipper_gacha.py:68]: 我用的WSGI會fork出兩個worker,全域變數有可能在兩個worker上造成資料不同部

File: flipper_gacha.py [hunk 2]
```diff
@@ -58,126 +64,167 @@ def limit_key_func():
...
+    return character_report_temp[pool]
+
+
+def check_pool(pool):
+    if pool not in pool_data_detal.keys() or pool is None or pool == "":
+        return list(pool_data_detal)[0]
+    return pool
+
+
+@app.errorhandler(404)
```
Comment (Zhen-Bo) [inline flipper_gacha.py:136]: 由於CDN是用cloudflare,所以這個func應該不會起作用(不會顯示回傳的error)

File: vue.config.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        opts[0].filename = '../templates/index.html';
+        return opts;
+      });
+
+    }
+  },
+  devServer: {
+    proxy: {
+      '/wf': {
+        target: 'http://192.168.100.147:5000/',
```
Comment (Zhen-Bo) [inline vue.config.js:17]: target在正式上線應該是要在port8001

File: flipper_gacha.py [hunk 4]
```diff
@@ -34,6 +39,7 @@
     MYSQL_PASSWORD=os.getenv("DB_PASS"),
     MYSQL_DB=os.getenv("DB_NAME"),
     MYSQL_CURSORCLASS="DictCursor",
+    DATA_FOLDER=os.getenv("DATA_FOLDER"),
```
Comment (Zhen-Bo) [inline flipper_gacha.py:42]: 這個DATA_FOLDER是?我沒有對應的路徑

File: flipper_gacha.py [hunk 5]
```diff
@@ -18,11 +19,15 @@
...
 import flask_limiter
 from gacha import gacha_pool
+from datetime import date
 
 load_dotenv()
 
+# vue build file
+app = Flask(__name__,
+            static_folder="./dist",
+            template_folder="./templates")
```
Comment (Zhen-Bo) [inline flipper_gacha.py:29]: 好像沒看到有dist,然後template_folder應該是./public?

File: flipper_gacha.py [hunk 6]
```diff
@@ -58,126 +64,167 @@ def limit_key_func():
...
+        rs = cursor.fetchall()
+        for row in rs:
+            character_list_temp[row['dev_id']] = row
+        cursor.close()
+    return character_list_temp[name]
+
+
+# Add character temp
+character_report_temp = {}
+last_run_time = {}
```
Comment (Zhen-Bo) [inline flipper_gacha.py:85]: 同上,全域變數可能會有問題

**Review by Zhen-Bo — CHANGES_REQUESTED** [review]
暫時只有這樣,因為啟動不了,所以剩下的沒辦法去測試&處理

**Comment by Zhen-Bo** [comment]
在npm install時會出現
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/webpack-dev-server/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

audited 1343 packages in 11.358s

91 packages are looking for funding
  run `npm fund` for details

found 7 moderate severity vulnerabilities
  run `npm audit fix` to fix them, or `npm audit` for details

無法在ubuntu系統完成安裝

File: vue.config.js (hunk 3 above)
Comment (bell-rabbit) [reply vue.config.js:17]: devServer 只是用在開發的設定，不能直接在PRD使用

File: flipper_gacha.py (hunk 4 above)
Comment (bell-rabbit) [reply flipper_gacha.py:42]: 由於static_folder轉了地方。DATA_FOLDER 是你的JSON.

VUE_APP_RESOURCE_URL=https://raw.githubusercontent.com/Zhen-Bo/flipper_gacha_simulator/main
DATA_FOLDER=./static

File: flipper_gacha.py (hunk 5 above)
Comment (bell-rabbit) [reply flipper_gacha.py:29]: 使用 `npm build` 來建立VUE PRD的dist及index.html ("./templates")

Comment (Zhen-Bo) [reply flipper_gacha.py:29]: 主要是可能你用的套件沒法安裝在ubuntu導致npm install會失敗
還是說直接build就好?

**Comment by bell-rabbit** [comment]
> 在npm install時會出現 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"}) npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"}) npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/webpack-dev-server/node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
> 
> audited 1343 packages in 11.358s
> 
> 91 packages are looking for funding run `npm fund` for details
> 
> found 7 moderate severity vulnerabilities run `npm audit fix` to fix them, or `npm audit` for details
> 
> 無法在ubuntu系統完成安裝

這WARN一直會出的，你可以無視他。
直接 `npm run serve`

File: flipper_gacha.py (hunk 1 above)
Comment (bell-rabbit) [reply flipper_gacha.py:68]: 不會不同步的， 這是個code table。
你要更改code table 的資料重開一下就好

File: flipper_gacha.py (hunk 6 above)
Comment (bell-rabbit) [reply flipper_gacha.py:85]: 同上。 character_report_temp 只是5分鐘不會再行多次。

File: flipper_gacha.py (hunk 2 above)
Comment (bell-rabbit) [reply flipper_gacha.py:136]: Cloudflare 可以回傳404 error


image

Comment (Zhen-Bo) [reply flipper_gacha.py:136]: 不過不會顯示error message吧?只會是cloudflare的錯誤頁面

File: flipper_gacha.py (hunk 5 above)
Comment (bell-rabbit) [reply flipper_gacha.py:29]: `npm run serve` 可以?

**Comment by Zhen-Bo** [comment]
> > 在npm install時會出現 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"}) npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"}) npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/webpack-dev-server/node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
> > audited 1343 packages in 11.358s
> > 91 packages are looking for funding run `npm fund` for details
> > found 7 moderate severity vulnerabilities run `npm audit fix` to fix them, or `npm audit` for details
> > 無法在ubuntu系統完成安裝
> 
> 這WARN一直會出的，你可以無視他。 直接 `npm run serve`

現在更尷尬地來了,我租借的vps是1vcpu+1Gram的,只要我一執行npm run build vps會直接卡死得手動重啟才能正常

File: flipper_gacha.py (hunk 5 above)
Comment (bell-rabbit) [reply flipper_gacha.py:29]: 一定會出現的

> PS C:\bell-rabbit\flipper_gacha_simulator> npm install
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules\watchpack-chokidar2\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules\webpack-dev-server\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

>audited 1343 packages in 5.286s

>91 packages are looking for funding
 > run `npm fund` for details

>found 7 moderate severity vulnerabilities
>  run `npm audit fix` to fix them, or `npm audit` for details

Comment (Zhen-Bo) [reply flipper_gacha.py:29]: ERROR  Failed to compile with 4 errors                                                                                                                    4:34:08 PM

This dependency was not found:

* ..\assets\bg.jpg in ./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"28508784-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/views/Flipper.vue?vue&type=template&id=0a314c46&scoped=true&

To install it, you can run: npm install --save ..\assets\bg.jpg


These relative modules were not found:

* ./undefined/static/image/rarity/star3.png in ./node_modules/css-loader/dist/cjs.js??ref--8-oneOf-1-1!./node_modules/vue-loader/lib/loaders/stylePostLoader.js!./node_modules/postcss-loader/src??ref--8-oneOf-1-2!./node_modules/sass-loader/dist/cjs.js??ref--8-oneOf-1-3!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/components/CharacterIcon.vue?vue&type=style&index=0&id=266f4242&lang=scss&scoped=true&
* ./undefined/static/image/rarity/star4.png in ./node_modules/css-loader/dist/cjs.js??ref--8-oneOf-1-1!./node_modules/vue-loader/lib/loaders/stylePostLoader.js!./node_modules/postcss-loader/src??ref--8-oneOf-1-2!./node_modules/sass-loader/dist/cjs.js??ref--8-oneOf-1-3!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/components/CharacterIcon.vue?vue&type=style&index=0&id=266f4242&lang=scss&scoped=true&
* ./undefined/static/image/rarity/star5.png in ./node_modules/css-loader/dist/cjs.js??ref--8-oneOf-1-1!./node_modules/vue-loader/lib/loaders/stylePostLoader.js!./node_modules/postcss-loader/src??ref--8-oneOf-1-2!./node_modules/sass-loader/dist/cjs.js??ref--8-oneOf-1-3!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/components/CharacterIcon.vue?vue&type=style&index=0&id=266f4242&lang=scss&scoped=true&

**Comment by bell-rabbit** [comment]
> > > 在npm install時會出現 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"}) npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"}) npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/webpack-dev-server/node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
> > > audited 1343 packages in 11.358s
> > > 91 packages are looking for funding run `npm fund` for details
> > > found 7 moderate severity vulnerabilities run `npm audit fix` to fix them, or `npm audit` for details
> > > 無法在ubuntu系統完成安裝
> > 
> > 
> > 這WARN一直會出的，你可以無視他。 直接 `npm run serve`
> 
> 現在更尷尬地來了,我租借的vps是1vcpu+1Gram的,只要我一執行npm run build vps會直接卡死得手動重啟才能正常

用你自己的電腦來build. 把build 好的`./dist`及 `./templates/index.html` 放上去

File: flipper_gacha.py (hunk 5 above)
Comment (Zhen-Bo) [reply flipper_gacha.py:29]: 這個是npm run serve
build則會直接讓vps卡死

Comment (bell-rabbit) [reply flipper_gacha.py:29]: 你自己的電腦來build.

.env 要加入新的設定
VUE_APP_RESOURCE_URL=https://raw.githubusercontent.com/Zhen-Bo/flipper_gacha_simulator/main
DATA_FOLDER=./static

File: flipper_gacha.py (hunk 2 above)
Comment (bell-rabbit) [reply flipper_gacha.py:136]: 會出。 這是個apache 404頁。

image

Comment (bell-rabbit) [reply flipper_gacha.py:136]: https://www.paverschlev.link/wf/flipper/34
你可以去看看

File: flipper_gacha.py (hunk 5 above)
Comment (bell-rabbit) [reply flipper_gacha.py:29]: npm run serve 是同來local dev 用的

Comment (bell-rabbit) [reply flipper_gacha.py:29]: https://book.vue.tw/CH3/3-3-vue-dev-server-build.html

**Comment by Zhen-Bo** [comment]
> > > > 在npm install時會出現 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"}) npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"}) npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/webpack-dev-server/node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
> > > > audited 1343 packages in 11.358s
> > > > 91 packages are looking for funding run `npm fund` for details
> > > > found 7 moderate severity vulnerabilities run `npm audit fix` to fix them, or `npm audit` for details
> > > > 無法在ubuntu系統完成安裝
> > > 
> > > 
> > > 這WARN一直會出的，你可以無視他。 直接 `npm run serve`
> > 
> > 
> > 現在更尷尬地來了,我租借的vps是1vcpu+1Gram的,只要我一執行npm run build vps會直接卡死得手動重啟才能正常
> 
> 用你自己的電腦來build. 把build 好的`./dist`及 `./templates/index.html` 放上去

 ERROR  Failed to compile with 1 error                                                                                                                                          上午12:53:03
 error  in ./src/components/CharacterIcon.vue?vue&type=style&index=0&id=fe104aec&lang=scss&scoped=true&

Syntax Error: ModuleNotFoundError: Module not found: Error: Can't resolve './undefined/static/image/rarity/star3.png' in 'C:\Users\Paver\Documents\GitHub\flipper_gacha_simulator\src\components'


 @ ./src/components/CharacterIcon.vue?vue&type=style&index=0&id=fe104aec&lang=scss&scoped=true& 1:0-535 1:0-535
 @ ./src/components/CharacterIcon.vue
 @ ./node_modules/cache-loader/dist/cjs.js??ref--12-0!./node_modules/thread-loader/dist/cjs.js!./node_modules/babel-loader/lib!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/components/CharacterReport.vue?vue&type=script&lang=js&
 @ ./src/components/CharacterReport.vue?vue&type=script&lang=js&
 @ ./src/components/CharacterReport.vue
 @ ./node_modules/cache-loader/dist/cjs.js??ref--12-0!./node_modules/thread-loader/dist/cjs.js!./node_modules/babel-loader/lib!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/views/Report.vue?vue&type=script&lang=js&
 @ ./src/views/Report.vue?vue&type=script&lang=js&
 @ ./src/views/Report.vue
 @ ./src/router/index.js
 @ ./src/main.js
 @ multi ./src/main.js

 ERROR  Build failed with errors.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! flipper_gacha_simulator@0.1.0 build: `vue-cli-service build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the flipper_gacha_simulator@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

在自己電腦build的時候出error
LOG:

18 verbose node v14.18.0
19 verbose npm  v6.14.15
20 error code ELIFECYCLE
21 error errno 1
22 error flipper_gacha_simulator@0.1.0 build: `vue-cli-service build`
22 error Exit status 1
23 error Failed at the flipper_gacha_simulator@0.1.0 build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Comment by bell-rabbit** [comment]
> > > > > 在npm install時會出現 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"}) npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"}) npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/webpack-dev-server/node_modules/fsevents): npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
> > > > > audited 1343 packages in 11.358s
> > > > > 91 packages are looking for funding run `npm fund` for details
> > > > > found 7 moderate severity vulnerabilities run `npm audit fix` to fix them, or `npm audit` for details
> > > > > 無法在ubuntu系統完成安裝
> > > > 
> > > > 
> > > > 這WARN一直會出的，你可以無視他。 直接 `npm run serve`
> > > 
> > > 
> > > 現在更尷尬地來了,我租借的vps是1vcpu+1Gram的,只要我一執行npm run build vps會直接卡死得手動重啟才能正常
> > 
> > 
> > 用你自己的電腦來build. 把build 好的`./dist`及 `./templates/index.html` 放上去
> 
> ERROR Failed to compile with 1 error 上午12:53:03 error in ./src/components/CharacterIcon.vue?vue&type=style&index=0&id=fe104aec&lang=scss&scoped=true&
> 
> Syntax Error: ModuleNotFoundError: Module not found: Error: Can't resolve './undefined/static/image/rarity/star3.png' in 'C:\Users\Paver\Documents\GitHub\flipper_gacha_simulator\src\components'
> 
> @ ./src/components/CharacterIcon.vue?vue&type=style&index=0&id=fe104aec&lang=scss&scoped=true& 1:0-535 1:0-535 @ ./src/components/CharacterIcon.vue @ ./node_modules/cache-loader/dist/cjs.js??ref--12-0!./node_modules/thread-loader/dist/cjs.js!./node_modules/babel-loader/lib!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/components/CharacterReport.vue?vue&type=script&lang=js& @ ./src/components/CharacterReport.vue?vue&type=script&lang=js& @ ./src/components/CharacterReport.vue @ ./node_modules/cache-loader/dist/cjs.js??ref--12-0!./node_modules/thread-loader/dist/cjs.js!./node_modules/babel-loader/lib!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/views/Report.vue?vue&type=script&lang=js& @ ./src/views/Report.vue?vue&type=script&lang=js& @ ./src/views/Report.vue @ ./src/router/index.js @ ./src/main.js @ multi ./src/main.js
> 
> ERROR Build failed with errors. npm ERR! code ELIFECYCLE npm ERR! errno 1 npm ERR! flipper_gacha_simulator@0.1.0 build: `vue-cli-service build` npm ERR! Exit status 1 npm ERR! npm ERR! Failed at the flipper_gacha_simulator@0.1.0 build script. npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> 在自己電腦build的時候出error LOG:
> 
> 18 verbose node v14.18.0 19 verbose npm v6.14.15 20 error code ELIFECYCLE 21 error errno 1 22 error flipper_gacha_simulator@0.1.0 build: `vue-cli-service build` 22 error Exit status 1 23 error Failed at the flipper_gacha_simulator@0.1.0 build script. 23 error This is probably not a problem with npm. There is likely additional logging output above. 24 verbose exit [ 1, true ]

.env 沒加上新的key 吧

**Comment by Zhen-Bo** [comment]
build完後把dict跟index.html放到對應的地方後還有要做什麼嗎?
我看畫面整個都是白的沒東西,也沒有跳出noscript的警告
body
`<body><noscript><strong>We're sorry but flipper_gacha_simulator doesn't work properly without JavaScript enabled. Please enable it to continue.</strong></noscript><div id="app"></div><script src="/js/chunk-vendors.2de79c06.js"></script><script src="/js/app.4fbdecff.js"></script><script defer="" src="https://static.cloudflareinsights.com/beacon.min.js" data-cf-beacon="{&quot;rayId&quot;:&quot;69775f667fac0ca7&quot;,&quot;version&quot;:&quot;2021.9.0&quot;,&quot;r&quot;:1,&quot;token&quot;:&quot;0d253e5e3cf940f885a857f4aca9fa94&quot;,&quot;si&quot;:100}"></script>
</body>`

**Comment by Zhen-Bo** [comment]
我目前把網站架設到/wf_test上面測試,不過我發現只要輸入https://www.paverschlev.link/wf_test後
他會自動把網址改寫成https://www.paverschlev.link/flipper
在 `/` 下面我有另一個APP再跑,所以會導致刷新後會變成404
要在哪邊才能修改會重寫網址的code
`/` ->一個APP
`/wf_test` ->測試用,也是單獨一個APP
`/wf`->現在的抽卡程式APP

**Comment by bell-rabbit** [comment]
> 我目前把網站架設到/wf_test上面測試,不過我發現只要輸入[https://www.paverschlev.link/wf_test後](https://www.paverschlev.link/wf_test%E5%BE%8C) 他會自動把網址改寫成https://www.paverschlev.link/flipper 在 `/` 下面我有另一個APP再跑,所以會導致刷新後會變成404 要在哪邊才能修改會重寫網址的code `/` ->一個APP `/wf_test` ->測試用,也是單獨一個APP `/wf`->現在的抽卡程式APP

由於我不知道你會放在那網址
你要手動設定一下 加上 /wf_test
再重新build 一次

`src/router/index.js`
`templates/`

**Comment by Zhen-Bo** [comment]
image
想問下最下面那個footer可不可以把額外兩邊的白色(框起來的部分)取消掉?
image
然後顯示的部分那個container可以把mt-2的class拿掉嗎?不然margin-top :8px是important會一直有空格
再來我覺得nav留下卡池切換的部分就好,那個彈射抽卡模擬的那條可以拿掉

自己的想像情況:
剛好是100vh
image

**Comment by Zhen-Bo** [comment]
剩下的我明天處理,先去睡了,反正明天應該會先上線,出事了再慢慢修就好

**Comment by Zhen-Bo** [comment]
感謝修改,不過這幾天都有排面試要準備所以可能要等之後有空才能更新Orz

**Comment by bell-rabbit** [comment]
已更新
image

假如你要把圖片分開來放 要加入這個新的file
`flipper_gacha_simulator\src\assets\bg.jpg`

**Comment by Zhen-Bo** [comment]
> 已更新 image
> 
> 假如你要把圖片分開來放 要加入這個新的file `flipper_gacha_simulator\src\assets\bg.jpg`

圖片分開放是啥意思,有點不太懂@@
然後你是說在build的時候就要有bg.jpg在assets這個資料夾內?我記得那個好像是個空白圖片...嗎?

**Comment by Zhen-Bo** [comment]
喔喔沒事了,我看到了

**Review by Zhen-Bo — APPROVED** [review]
LGTM

**Pull request merged**